### PR TITLE
Upgrade on channels required

### PR DESCRIPTION
**Pull request by pthegner**
# Description

This PR add some check on channels in order to report an error if a required channel is not send

## Issue

Issue link: BOT-1089

## How to test

Here are the following steps to test this pull request:

- create or update an impact with no messages entry (with a client who have at least one channel with required=true)
- you should have a 400 response from the API with a distinctive message

### CODE REVIEW

File: chaos/db_helper.py [hunk 1]
```diff
@@ -210,13 +211,18 @@ def manage_message(impact, json, client_id):
                 impact.insert_message(message)
                 manage_message_meta(message, message_json)
                 messages_db[message.channel_id] = message
+    elif channels_required:
+        raise exceptions.InvalidJson('Channel{} {} {} required.'.format(
+            's' if len(channels_required) > 1 else '',
+            ', '.join(list(ch.id for ch in channels_required)),
+            'are' if len(channels_required) > 1 else 'is'
+        ))
```
Comment (Fahmus) [inline chaos/db_helper.py]: raise exceptions.InvalidJson('Channel{} with id{} {} {} required.'.format(
            's' if len(channels_required) > 1 else '',
            's' if len(channels_required) > 1 else '',
            ', '.join(list(ch.id for ch in channels_required)),
            'are' if len(channels_required) > 1 else 'is'
        ))

File: tests/features/create-disruption-with-impacts.feature [hunk 2]
```diff
@@ -240,3 +240,77 @@ Feature: Create Disruption and impacts
...
+        I fill in header "X-Customer-Id" with "5"
+        I fill in header "X-Coverage" with "jdr"
+        I fill in header "Authorization" with "d5b0148c-36f4-443c-9818-1f2f74a00be0"
+        When I post to "/disruptions" with:
+        """
+        {"reference": "foo","contributor": "contrib1","cause": {"id": "7ffab230-3d48-4eea-aa2c-22f8680230b6"},"publication_period":{"begin":"2018-09-11T13:50:00Z","end":"2018-12-31T16:50:00Z"},"impacts": [{"severity": {"id": "7ffab232-3d48-4eea-aa2c-22f8680230b6"},"objects": [{"id": "network:JDR:1","type": "network"}],"application_periods": [{"begin": "2014-04-29T16:52:00Z","end": "2014-06-22T02:15:00Z"},{"begin": "2014-04-29T16:52:00Z","end": "2014-05-22T02:15:00Z"}]}]}
+        """
+        Then the status code should be "400"
+        And the header "Content-Type" should be "application/json"
+        And the field "error.message" should be "Channels 5ffab230-3d48-4eea-aa2c-22f8680230b6, 5ffab232-3d48-4eea-aa2c-22f8680230b6 are required."
```
Comment (Fahmus) [inline tests/features/create-disruption-with-impacts.feature]: new line?

Comment (pthegner) [reply tests/features/create-disruption-with-impacts.feature]: What do you mean by that ?

Comment (dvdn) [inline tests/features/create-disruption-with-impacts.feature]: I think he means line return at the end of file

File: chaos/db_helper.py [hunk 3]
```diff
@@ -299,7 +305,7 @@ def create_or_update_impact(disruption, json_impact, navitia, impact_id=None):
             # we insert this object in the table pt_object
             if pt_object_json["type"] == 'line_section':
                 try:
-                    ptobject = fill_and_add_line_section(navitia, impact_bd.id, all_objects, pt_object_json)
+                    ptobject = fill_and_add_line_section(navitia, all_objects, pt_object_json)
                 except exceptions.ObjectUnknown as xxx_todo_changeme:
```
Comment (dvdn) [inline chaos/db_helper.py]: `xxx_todo_changeme` :joy:

File: tests/features/create-disruption-with-impacts.feature (hunk 2 above)
Comment (pthegner) [reply tests/features/create-disruption-with-impacts.feature]: Is it mandatory ? because before there was no line at the bottom.

Comment (pthegner) [reply tests/features/create-disruption-with-impacts.feature]: I can do that but if i do, i think i'll have to change all the feature files 🤣

File: chaos/db_helper.py [hunk 4]
```diff
@@ -210,13 +211,18 @@ def manage_message(impact, json, client_id):
                 impact.insert_message(message)
                 manage_message_meta(message, message_json)
                 messages_db[message.channel_id] = message
+    elif channels_required:
```
Comment (dvdn) [inline chaos/db_helper.py]: Because 'if' condition can be true when empty `messages[ ]`, it never will come inside this 'elif'

Comment (pthegner) [reply chaos/db_helper.py]: and ??

Comment (pthegner) [reply chaos/db_helper.py]: The fix with this PR is to return an error when the messages part is not send. I know that with my modification, the message is clear for this part but if i have to do a refacto on the other part...
we should see that in a call.

File: tests/features/create-disruption-with-impacts.feature (hunk 2 above)
Comment (Fahmus) [reply tests/features/create-disruption-with-impacts.feature]: Yes! that's what I mean :)

File: chaos/db_helper.py [hunk 5]
```diff
@@ -190,31 +190,37 @@ def clean_message(msg, type=''):
         msg.text = msg.text.replace('\r\n', ' ')
 
 def manage_message(impact, json, client_id):
+    if 'messages' not in json:
```
Comment (dvdn) [inline chaos/db_helper.py]: I'm not comfortbale with that, but perhaps a comment can do the trick
```suggestion
   #same management for every cases as 'messages' is not required
    if 'messages' not in json:
```

**Pull request merged**